core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -48,6 +48,29 @@ import scala.collection.mutable.ArrayBuffer
 class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends Logging {
   import KafkaZkClient._
 
+  def createSequentialPersistentPath(path: String, data: String = ""): String = {
+    val createRequest = CreateRequest(path, data.getBytes("UTF-8"), acls(path), CreateMode.PERSISTENT_SEQUENTIAL)
+    val createResponse = retryRequestUntilConnected(createRequest)
+    createResponse.path
+  }
+
+  def leaderAndIsrZkData(leaderAndIsr: LeaderAndIsr, controllerEpoch: Int): String = {
+    Json.encode(Map("version" -> 1, "leader" -> leaderAndIsr.leader, "leader_epoch" -> leaderAndIsr.leaderEpoch,
+                    "controller_epoch" -> controllerEpoch, "isr" -> leaderAndIsr.isr))
+  }
+
+  def readDataMaybeNull(path: String): (Option[String], Stat) = {
```

Comment:
With this, methods like getPreferredReplicaElection() and getControllerId, etc could potentially be simplified by reusing this method.